ee/modules/110-istio/hooks/discovery_revisions.go

```diff
@@ -37,15 +37,24 @@ func revisionsDiscovery(input *go_hook.HookInput, dc dependency.Container) error
 	}
 
 	var globalVersion string
+
+	// globalVersion is set in CM — use it
 	if input.ConfigValues.Exists("istio.globalVersion") {
-		// globalVersion is set in CM — use it
 		globalVersion = input.ConfigValues.Get("istio.globalVersion").String()
 
 		if !internal.Contains(supportedVersions, globalVersion) {
 			unsupportedVersions = append(unsupportedVersions, globalVersion)
 		}
-	} else {
-		// let's check the d8-istio/istiod Service's istio.deckhouse.io/global-version annotation
+	}
+
+	// Check if istio.internal.globalRevision is already set to prevent excess API calls.
+	if globalVersion == "" {
+		globalVersion = input.Values.Get("istio.internal.globalRevision").String()
```

Comment:
version != revision, ill review the whole hook.

Reply:
Nested ifs before:

```
if config istio.globalVersion {
  globalVersion = config istio.globalVersion
} else {
   Get annotations from Service
   if ok {
     globalVersion = annotation "istio.deckhouse.io/global-version"
   } else {
     globalVersion = values istio.globalVersion
   }
}

globalRevision = Convert globalVersion to revision
...
Save globalRevision into istio.internal.globalRevision
```

Now it is more linear:

```
if config istio.globalVersion {
  globalVersion = config istio.globalVersion
}

// Restore from internal values
if globalVersion is empty {
  globalVersion = values istio.internal.globalRevision
}

if globalVersion still empty {
  Get annotations from Service
   if ok {
     globalVersion = annotation "istio.deckhouse.io/global-version"
   }
}

if globalVersion still empty {
     globalVersion = values istio.globalVersion
}

globalRevision = Convert globalVersion to revision
...
Save globalRevision into istio.internal.globalRevision
```

Note: `VersionToRevision` detects if input string is already revision.